index.js

```diff
@@ -0,0 +1,372 @@
+                        algorithm: keys[keyId].algorithm,
+                        expires: keys[keyId].expires
+                    };
+
+                    if (maxDate > keys[keyId].expires) {
+                        return false;
+                    }
+
+                    return true;
+                });
+
+                return output;
+            }
+        }
+    };
+};
+
+/**
+ * Creates a Remote auth service
+ * @method loadRemote
+ * @param  {Object}  authConfig           Input for setting up a remote version
+ * @param  {String}  authConfig.remoteUri URI of remote server
+ * @param  {String}  authConfig.timeout   How long to wait for API response
+ * @return {Object}                       Contains keys, apis, etc.
+ */
+internals.loadRemote = (authConfig) => {
+    const mutex = semaphore(1);
+    let keys = {};
+
+    return {
+        /**
+         * Load the key data (from the remote server if needed)
+         * @method getKey
+         * @param  {String}   keyId Key you are looking for
+         * @param  {Function} next  Function to call when done (error, key)
+         */
+        getKey: (keyId, next) => {
+            const nextSem = (err, result) => {
+                mutex.leave();
+                next(err, result);
+            };
+
+            // Prevent multiple calls to the remote URI
+            mutex.take(() => {
+                if (keys[keyId]) {
+                    return nextSem(null, keys[keyId]);
+                }
+
+                return request({
+                    url: authConfig.remoteUri,
+                    headers: {
+                        'user-agent': `${pkg.name}@${pkg.version}`,
+                        'hapi-auth-requested-key': keyId
+                    },
+                    json: true,
+                    timeout: authConfig.timeout
+                }, (reqError, response, body) => {
+                    if (reqError || response.statusCode !== 200) {
+                        const prevError = reqError ||
+                            new Error(`Status Code: ${response.statusCode}`);
+
+                        return nextSem(prevError);
+                    }
+
+                    keys = {};
+                    body.forEach((key) => {
+                        keys[guid(key.public)] = key;
+                    });
+
+                    if (keys[keyId]) {
+                        return nextSem(null, keys[keyId]);
+                    }
+
+                    return nextSem(new Error('Specified Key Id not found'));
+                });
+            });
+        }
+    };
+};
+
+/**
+ * Configures the authentication plugin of Hapi
+ * @method implementation
+ * @param  {HapiServer} server           HapiServer we are attaching the auth service to
+ * @param  {Object}     config           Plugin configuration
+ * @param  {Array}      config.keys      List of private keys to sign JWTs with
+ * @param  {String}     config.maxAge    Maximum age for signing JWTs
```

Comment:
missing config.issuer?

Reply:
and I'm not sure how to indicate here that it could be either (keys, maxAge, issuer) OR (remoteUri, timeout)?

Reply:
Fixed (and I don't know how)